core/meta/src/TClass.cxx

```diff
@@ -5464,29 +5464,24 @@ TClass *TClass::LoadClassCustom(const char *requestedname, Bool_t silent)
 
 void TClass::LoadClassInfo() const
 {
-   R__LOCKGUARD(gInterpreterMutex);
+   // Load class info if not already loaded and auto-parsing is not suspended
+   if (!fClassInfo && !gInterpreter->IsAutoParsingSuspended()) {
+      R__LOCKGUARD(gInterpreterMutex);
 
-   // If another thread executed LoadClassInfo at about the same time
-   // as this thread return early since the work was done.
-   if (!fCanLoadClassInfo) return;
```

Comment:
The comment is not fully accurate.  There is more semantic to fCanLoadClassInfo, it also cover the case where fClassInfo is null *but* the TClass knows it will *not* be able to load it (and thus there is not point in wasting CPU re-finding out that information).

And also fCanLoadClassInfo is an atomic.